### ISSUE TITLE
Please Verify Local Data! Piece #xxx is corrupt

### ISSUE THREAD

**Issue by RuralHunter** [transmission/transmission#820]
From time to time I see the error on my already completed torrents and then they stop seeding. I had to restart them manually and this is very annoying. My download files are stored in a samba path but there should be no problem accessing it all the time. There is also not any local modification to the files. Besides, I haven't seen any problem with other bt clients with the same situation. I think transmission at least can try to restart those torrents automatically even there is really some corrupt somehow.

**Comment by herebcs** [comment]
I also experienced the same occasional errors popping up. In my case the temporary download folder was on an internal SSD and the files were move to a USB hard drive after downloading. Transmission might screw something up during the relocation of the files.
I solved this by setting up a script to initiate an extra round of verification when a torrent is completed.

**Comment by ckerr** [comment]
> I think transmission at least can try to restart those torrents automatically even there is really some corrupt somehow.

I don't understand how this would work in practice. What should Transmission do if a peer requests a piece that Transmission thinks is corrupt? What's the purpose of staying seeding when the client thinks there are errors?

**Comment by ctscobra** [comment]
I have this issue over NFS and it is a big problem . With growing number of downloads i can get it every few minutes on random downloads where either says "stale file" or "corrupt file" just pressing "resume now" it is fixed until some other gets the same issue . It is on Linux " 2.94-1ubuntu1~18.04.1"

**Comment by RuralHunter** [comment]
Yes, exactly same with me. I think it maybe the disk read/write timeout is set to a value too short.

**Comment by mauron85** [comment]
I've seen this yesterday. My torrent was just being downloaded. It stop downloading because this error. This is unfortunate when you expect to download files while you sleep and in the morning there is this error instead. I would expect transmission to finish download and display message to the user, so he/her can verify local data on finished download.

**Comment by liuyix** [comment]
same to me

**Comment by itskando** [comment]
+1

**Comment by Zbysekz** [comment]
same problem here

**Comment by Zbysekz** [comment]
"Verify data" wasn't working for me, it does nothing. But after transmission restart "Verify data" works and it corrects the corrupt problem.

**Comment by itskando** [comment]
+1 - same.

**Comment by ahsanhussain** [comment]
+1

**Comment by ctscobra** [comment]
Will this be acknowledged by devs ? I get this on 40+ torrents at least 30 times per day for months now (since i started using transmission)

**Comment by ctscobra** [comment]
so I got sick and tired of this issue and added this to crontab , it will list all failed downloads every 5 minutes and start them:

`
*/5 * * * * for i in $(transmission-remote -n 'username:password' -l | grep Stopped | awk '{print $1}' | grep \* | sed 's/\*//g') ; do transmission-remote -n 'username:password' -t$i --start ; done
`

**Comment by mvv588** [comment]
I can confirm this issue with version 2.84, the files are accessed over cifs, only transmission accesses the files. Sometimes this happens when downloading, most of the time it is when seeded. After some minutes the file goes from 100% to 99.x% and paused with the triangle and ! symbol. 

Click resume and it downloads to 100%, seeds and most times goes to 99.x% again after some time. All the while other files at seeding just fine at the same time. In the log I find the error message in the title of this bug report.

The problematic files are actually fine and usable no errors from other applications.

**Comment by skewty** [comment]
This is affecting me as well.  As above, I think it has to do with small timeout window somewhere or something similar.  When I reduced the likelihood of a slow random read (by reducing the number of seeding torrents, the problem went away).

**Comment by devesh** [comment]
Transmission should redownload corrupt chunks.

**Comment by lordkitsuna** [comment]
I am also having these issues, as is one of my friends. We thought something was wrong with our storage before finding this thread. Seems pretty clear many people are having this problem. can we at least get some kind of confirmation the devs are aware of the issue and not just ignoring it?

**Comment by touche112** [comment]
Same issue here.  Ubuntu 18.04 + Transmission 2.92 with CIFS storage.

**Comment by Laca81** [comment]
Same here. Ubuntu 18.04 + Transmission 2.92

**Comment by tcard** [comment]
Inspired by @ctscobra, since I don't have (nor want) `transmission-remote`, I wrote this AppleScript that resumes all transfers every second.

```applescript
repeat
	tell application "System Events"
		tell process "Transmission"
			click menu item "Resume All" of menu "Transfers" of menu bar 1
		end tell
	end tell
	delay 1
end repeat
```

Just paste that on Utilities/Script Editor, hit ▶️ and let it run in the background. (It will ask for accessibility permissions. Enable the Script Editor app when prompted.)

**Comment by pythoninthegrass** [comment]
> 
> 
> so I got sick and tired of this issue and added this to crontab , it will list all failed downloads every 5 minutes and start them:
> 
> `*/5 * * * * for i in $(transmission-remote -n 'username:password' -l | grep Stopped | awk '{print $1}' | grep \* | sed 's/\*//g') ; do transmission-remote -n 'username:password' -t$i --start ; done`

Thanks for posting this @ctscobra! While it works as-is, golfed it to just use `awk` and pushed out to every 30 minutes:
```*/30 * * * * for i in $(transmission-remote -n 'username:password' -l | awk '/Stopped/ {print $1}') ; do transmission-remote -n 'username:password' -t$i --start ; done```

**Comment by Laca81** [comment]
I fixed the problem by replacing the hdd. Hdsentinel said health is about 13%... There was a lot of badsector on it. After the replacement all downloads completes.

**Comment by vchimishuk** [comment]
Same issue for me on pretty simple configuration: SSD, no NFS shares or something, no files moving after download. Seeding torrent can be marked as corrupted after beeing seeded for a week.

**Comment by ctscobra** [comment]
> Same issue for me on pretty simple configuration: SSD, no NFS shares or something, no files moving after download. Seeding torrent can be marked as corrupted after beeing seeded for a week.

This could be becouse of trim or different cache manipulation that SSD's have . But conclusion is same , transmission is missing crucial functions.

**Comment by hamah99** [comment]
I've got the same problem with 2.94. Interestingly this issue only happens on a few of the torrents I'm seeding but it happens over and over on those same torrents. I've run First Aid in Disk Utility and that comes up clean and I don't seem to have any data integrity issues with any other application but Transmission. The external drive I'm using is a Drobo. 

I've updated to 3.00 now that it's out and will report back if there is any improvement or if the problem remains. A little frustrating that people have seen this issue for almost 1.5 yrs and nothing from the devs as to what the issue might be, how to provide further info on the problem, ...crickets

**Comment by shivdhar** [comment]
Same issue. It is rather amazing that there is a "Verify" button but not a "Fix torrent" button anywhere that re-downloads corrupt pieces. There isn't a way to do anything even if there are corrupt pieces, that I could find.

Or has anyone found a way, short of switching to another client like Tixati?

**Comment by andreygursky** [comment]
> Not sure how, is re-downloading corrupt pieces part of the "Verify" functionality

Considering a stopped torrent. "Verify" marks corrupted pieces as missing. Once you start it manually, the missing pieces will be downloaded.

**Comment by vchimishuk** [comment]
I've resolved my issue migration to HDD from SSD. Not sure how SSD may affects Transmission.

**Comment by KamikazeRaven** [comment]
I'm having a similar issue but not exactly the same.

It happens after I use "Set Location..." option to move the contents to other directories. Regardless of whether the download is completed, I often end up seeing a little bit of corruption. One of my 71GB download have 67.1MB corrupted, but it affected 80% of the files. All of the affected video files cannot be played.

Even though transmission shows there are corrupted files, it still seeds. I tried to "Verify Local Data", but it does not seem to be fixing the problem. After the verification process finished, the torrents just resumes to be seeded, and I still cannot play those video files. And transmission still shows the same amount of data is corrupted.

I'm running version 3.00 on FreeNAS with the webGUI. I'll let it run overnight and see if it fixes the problem.

**Comment by bassmadrigal** [comment]
This issue could likely be mitigated if transmission devs would add an option to verify completed torrents. Most torrent clients have this, and this is seriously making me think of switching to another client (which I don't want to do because I like all other aspects of it).

It doesn't happen extremely often for me, but maybe it happens more frequently than is caught by the client. I just had one pop up on a torrent I've been seeding for a month. I'm not sure if the corrupted piece was there from the start or something the client messed up after everything was completed (which if it is the latter, is a very serious issue). If it's the former, I might be moving completed torrents that have corrupt pieces, all because the devs refuse to add an option to verify torrents when they complete. It's sad that we have to switch to community scripts to add functionality that should be present in the program itself.

**Comment by hamah99** [comment]
Still seeing this issue with 3.00. All was going well for a number of months and I thought the problem was gone until I upgraded to Mac OS 11 (Big Sur).

**Comment by ctscobra** [comment]
I commented on this thread years ago now :D , I am verry pissed because i have countles of corrupted torrents from that era (before i did manual verify of every torrent before removing it from transmission).

**Comment by victorgrund** [comment]
I still see this all the time on 3.00 (bb6b5a062e).  It's killing me.

**Comment by TomArrow** [comment]
I have this same issue. Why in the world is this unfixed after 2 years? I am trying to download a huge torrent and it keeps erroring out. "Start Now" fixes it temporarily but that's not a solution. 

Can anyone else please recommend a tool that can handle very very large torrent files (think hundreds of thousands of files) and doesn't have this issue? This is important data to me that I must transfer and I need to be able to run it overnight.

**Comment by shivdhar** [comment]
Hi @TomArrow, you might want to try Tixati: https://www.tixati.com/download. It's has a rather retro UI, but works quite reliably in my experience.

**Comment by vchimishuk** [comment]
@TomArrow , please, share your setup (is it SSD or HDD, OS and other). Issue gone for me after migration to HDD.

**Comment by TomArrow** [comment]
> Hi @TomArrow, you might want to try Tixati: https://www.tixati.com/download. It has a rather retro UI, but works quite reliably in my experience.

Thanks, sadly I already tried that one and it failed to load the torrent too.

> @TomArrow , please, share your setup (is it SSD or HDD, OS and other). Issue gone for me after migration to HDD.

HDD in an external USB3 docking station.

**Comment by hamah99** [comment]
there was an Apple script posted on April 19 of last year by Tcard that resumes all transfers every second. That didn't work well for me as I needed enough time to redownload the corrupt bits. I extended the time to 60 seconds and that works pretty well with me only having to occasionally intercede manually. It's sad that this issue remains after all this time with zero response from the dev(s).

**Comment by bgdnlp** [comment]
Found this issue when searching for the message. I'm on MacOS Monterey, Transmission 3.00+ (0166b6e688). It fixed it after a manual Verify.

I quit Transmission and started again. This is probably not needed. Then, with the torrent selected in the list, went to Transfers -> Verify Local Data. It did, then it switched to Downloading, got the corrupted piece and the torrent is good now. It was still seeding while the error was showing.

The only things I would change is
- put Verify Local Data on right click menu
- Try to fix it automatically at least once, or when a bunch of new seeds show up, or something similar

Otherwise this seems to be fixed?

**Comment by hamah99** [comment]
Those seem like reasonable methods to manually resolve the issue. Problem is that there is no problem with the data. I've tried to find file corruption and/or bit rot and have been unable to do so. The fact that you have to intercede at all is unfortunate but not so much as there appears to not even be an acknowledgment that a problem exists at all after over 2 years. I appreciate all the work the devs and contributors do on this code, don't get me wrong. This thread is just a bitch session though as nothing constructive seems to come from it.

**Comment by bgdnlp** [comment]
> Problem is that there is no problem with the data. I've tried to find file corruption and/or bit rot and have been unable to do so.

Not to invalidate what you found, but in my case there really was corrupt data and it did get fixed after I ran the steps above.

Also, it's the first time I saw this message in years of using Transmission, which is why I had to search for it. But I'm not a heavy torrent user, so that doesn't really say much. The first post does have only 4 thumbs-up in 3 years though.

**Comment by ctscobra** [comment]
Issue is that Transmission is not doing recheck/verify automatically , i had thousands of corrupt downloads , never had those issues with utorrent/bittorrent since they always do a verify after finishing .

**Comment by ctscobra** [comment]
> > Problem is that there is no problem with the data. I've tried to find file corruption and/or bit rot and have been unable to do so.
> 
> Not to invalidate what you found, but in my case there really was corrupt data and it did get fixed after I ran the steps above.
> 
> Also, it's the first time I saw this message in years of using Transmission, which is why I had to search for it. But I'm not a heavy torrent user, so that doesn't really say much. The first post does have only 4 thumbs-up in 3 years though.

"it works for me" is not really a valid way of thinking . I did not have this issue until i scaled to NAS with 16 drives.

**Comment by koutsenko** [comment]
Same issue on Keenetic router's built-in transmission client

**Comment by Tao311** [comment]
Same issue 3.0 synology nas hdd

**Comment by rolex905** [comment]
Same issue here. Transmission 2.94 with USB hard disk.

**Comment by ShipOfUnraid** [comment]
+1 on this one, 2.94 on Unraid

**Comment by SilenceEagle** [comment]
Same issue, Transmission 3.0, `QNAP` NAS, HDD. But only on several new downloaded torrents, restarting works for a while, then errors occur again. The other torrents work always good.

**Comment by sukibaby** [comment]
Not sure if I should mention this here or elsewhere, but I did experience this:

> "Verify data" wasn't working for me, it does nothing. But after transmission restart "Verify data" works and it corrects the corrupt problem.


The corrupted data was **not** a problem with Transmission;  I had the file from the affected chunk open in another application while the torrent was still downloading, and edited it while the torrent was still downloading. So it was a **legitimate data corruption unrelated to Transmission**.

However, I did have to close the app and restart it before "Verify Local Data" in the right-click menu, or CTRL-V would have any effect. I tried both pausing and unpausing the affected torrent before ultimately restarting the app, where it then responded as expected to clicking "Verify Local Data".
